### PR TITLE
feat: use parentId in Spans

### PR DESCRIPTION
**Pull request by jahtalab**
minor fixes and refactoring

### CODE REVIEW

File: src/performance-monitoring/span-base.js [hunk 1]
```diff
@@ -2,6 +2,17 @@ const utils = require('../common/utils')
 class SpanBase {
   // context
 
+  constructor (name, type, options) {
+    if (!options) options = {}
+    this.name = name
+    this.type = type
+    this.options = options
```
Comment (vigneshshanmugam) [inline src/performance-monitoring/span-base.js]: nit: `options || {}` instead of line 6 to avoid reassignment

File: dev-utils/karma.js [hunk 2]
```diff
@@ -158,8 +158,8 @@ function prepareConfig (defaultConfig) {
       defaultConfig.plugins.push('karma-coverage')
 
       var babelPlugins =
-        defaultConfig.webpack.module.loaders[0].query.plugins ||
-        (defaultConfig.webpack.module.loaders[0].query.plugins = [])
+        defaultConfig.webpack.module.rules[0].options.plugins ||
```
Comment (vigneshshanmugam) [inline dev-utils/karma.js]: do we even use it somewhere?

Comment (jahtalab) [reply dev-utils/karma.js]: It's used in the line 163 to add `istanbul` which provides code coverage reports for us! 
You can run it with `npm run karma-coverage`

**Comment by jahtalab** [comment]
Thanks @vigneshshanmugam , I've made some changes!

**Pull request merged**